apisix/plugins/ua-restriction.lua

```diff
@@ -128,12 +144,13 @@ function _M.access(conf, ctx)
             return 403, { message = conf.message }
         end
     end
-    local match = MATCH_NONE
+    local match
+
     if type(user_agent) == "table" then
         for _, v in ipairs(user_agent) do
             if type(v) == "string" then
                 match = lrucache_useragent(v, conf, match_user_agent, v, conf)
-                if match > MATCH_ALLOW then
+                if (conf.denylist and not match) or (conf.allowlist and match) then
```

Comment:
The logic is very complicated, Why not add a check function for each of denylist and allowlist?
The function only returns `local PASS = 0` and `local DENY = 1`

Reply:
Updated.